### PR TITLE
POOL-705 Add StakePrizePool

### PR DESCRIPTION
**Pull request by ddesrosier**

### CODE REVIEW

File: lib/components/BuilderUI.jsx
Comment (chuckbergeron) [inline lib/components/BuilderUI.jsx:262]: same with this. Looks like sometimes it's moving useState args to new lines, and sometimes it's not?

File: lib/components/PoolTypeSelector.jsx
Comment (chuckbergeron) [inline lib/components/PoolTypeSelector.jsx:18]: Don't need `props.` here if we're pulling it out of the object on line 8 above.

File: lib/constants.js
Comment (chuckbergeron) [inline lib/constants.js:22]: Nice, I forgot about `Object.freeze()`

**Review by chuckbergeron — CHANGES_REQUESTED** [review]
Looks great! I think my only comments were around the formatter config.

File: lib/components/BuilderUI.jsx
Comment (ddesrosier) [reply lib/components/BuilderUI.jsx:262]: I think these are line length restrictions, we can adjust that

File: lib/components/BuilderForm.jsx
Comment (ddesrosier) [reply lib/components/BuilderForm.jsx:53]: We should be able to. I'll go through your comments & add rules for everything.

File: lib/components/BuilderForm.jsx
Comment (ddesrosier) [reply lib/components/BuilderForm.jsx:46]: agreed

File: lib/components/BuilderForm.jsx
Comment (ddesrosier) [reply lib/components/BuilderForm.jsx:56]: So we want
```
 <form 
    onSubmit={handleSubmit}
  >
        <div
           className='font-bold mb-8 py-2 text-lg sm:text-xl lg:text-2xl text-default'
        >
```
?

File: lib/components/BuilderForm.jsx
Comment (ddesrosier) [reply lib/components/BuilderForm.jsx:53]: Ah oh https://github.com/prettier/prettier/issues/2724

Comment (chuckbergeron) [reply lib/components/BuilderForm.jsx:53]: hmm, that's annoying. I wonder how hard it would be to patch prettier and submit a PR.

That or use something else like standard, and just have it behave similar to prettier.

File: lib/components/BuilderForm.jsx
Comment (chuckbergeron) [reply lib/components/BuilderForm.jsx:56]: Yep, that'd be awesome. It's helpful having the closing fragment on a new line to see where the opening tag ends.

File: lib/components/BuilderForm.jsx
Comment (chuckbergeron) [reply lib/components/BuilderForm.jsx:53]: Oh just saw the maintainers note about not accepting a PR for this. Well we can fork it and just run our own version of it.

**Comment by chuckbergeron** [comment]
oh shoot, don't merge this yet! I forgot to mention that when you push to the `master` branch with this and the reference app repo it deploys to netlify/prod. Which would normally be fine but if the reference app doesn't support Stake prize pools then we don't want these changes live yet.

File: lib/components/BuilderForm.jsx
Comment (ddesrosier) [reply lib/components/BuilderForm.jsx:53]: It looks like the parenthesis are just the default accepted way of formatting multi-line jsx. Standard is super opinionated and doesn't encourage customization. I'll look a little more tomorrow. Tbh I kinda like it now that I've read some examples of why it's the standard. Keeping the HTML tags indented properly is pretty.

File: lib/components/BuilderForm.jsx
Comment (ddesrosier) [reply lib/components/BuilderForm.jsx:56]: Looks like this isn't a simple toggle either. Prettier will only format to ^ if there's multiple props & the line reaches the max length.

**Pull request merged**
